tests/src/test/scala/com/nvidia/spark/rapids/GpuMultiFileReaderSuite.scala

```diff
@@ -74,4 +78,62 @@ class GpuMultiFileReaderSuite extends AnyFunSuite {
       assertResult(false)(multiFileReader.next())
     }
   }
+
+  def generateRandomString(length: Int): String = {
+    Iterator.fill(length)(Random.alphanumeric.head).mkString
+  }
+
+  def sqlExprToGenerateRandomString(length: Int, partitions: Int): String = {
+    val randomString = generateRandomString(length)
+    s"substring('$randomString', 1, floor(rand() * $partitions) + 1)"
+  }
+
+  def writeAndReadParquet(df: DataFrame, spark: SparkSession, outputFile: File,
+      partCols:String*): DataFrame = {
+    spark.conf.set("spark.rapids.sql.enabled", "false")
+    df.write.partitionBy(partCols:_*).parquet(outputFile.getCanonicalPath)
+
+    spark.conf.set("spark.rapids.sql.enabled", "true")
+    spark.read.parquet(outputFile.getCanonicalPath)
+  }
+
+  test("WIP: test column size exceeding cudf limit" +
+    " - single partition column, single partition value") {
+    val numRows = 10000
+    withTempPath { file =>
+      withGpuSparkSession(spark => {
+        val df = spark.range(numRows)
+          .withColumn("partCol", lit(generateRandomString(10)))
+        val resultAns = writeAndReadParquet(df, spark, file, "partCol").collect()
+        assert(resultAns.length == numRows)
+      })
+    }
+  }
+
+  test("WIP: test column size exceeding cudf limit" +
+    " - single partition column, multiple partition values") {
+    val numRows = 10000
+    withTempPath { file =>
+      withGpuSparkSession(spark => {
+        val df = spark.range(numRows)
+          .withColumn("partCol", expr(sqlExprToGenerateRandomString(20, 10)))
+        val resultAns = writeAndReadParquet(df, spark, file, "partCol").collect()
+        assert(resultAns.length == numRows)
+      })
+    }
+  }
+
+  test("WIP: test column size exceeding cudf limit" +
+    " - multiple partition columns, multiple partition values") {
+    val numRows = 10000
+    withTempPath { file =>
+      withGpuSparkSession(spark => {
+        val df = spark.range(numRows)
+          .withColumn("partCol", expr(sqlExprToGenerateRandomString(20, 10)))
+          .withColumn("partCol2", expr(sqlExprToGenerateRandomString(20, 10)))
```

Comment:
There should be a test that explicitly tests the case where one partition column triggers a split and then a subsequent partition column triggers an _earlier_ split (i.e.: has a wider value than the previous column).